src/comm.cc

```diff
@@ -200,11 +201,32 @@ comm_local_port(int fd)
     return F->local_addr.port();
 }
 
+/* Sets the IP_BIND_ADDRESS_NO_PORT sock option to enable reuse of Linux ephemeral ports */
+static inline void
+CommSetBindNoPort(int s)
+{
+    int bind_noport_flag = 1;
+#if _SQUID_LINUX_
+    if (setsockopt(s, IPPROTO_IP, IP_BIND_ADDRESS_NO_PORT, (char *) &bind_noport_flag, sizeof(int)) < 0) {
+        int xerrno = errno;
+        debugs(50, DBG_IMPORTANT, MYNAME << "setsockopt(IP_BIND_ADDRESS_NO_PORT) " << (bind_noport_flag?"ON":"OFF") << " for FD " << s << ": " << xstrerr(xerrno));
+    }
+#else
+    debugs(50, DBG_CRITICAL, MYNAME << "WARNING: setsockopt(IP_BIND_ADDRESS_NO_PORT) not supported on this platform");
+#endif
+}
+
 static Comm::Flag
 commBind(int s, struct addrinfo &inaddr)
 {
     ++ statCounter.syscalls.sock.binds;
 
+    /* If this is a socket for an outbound TCP session, set the flag for ephemeral source port reuse */
+    sockaddr_in* addr = (sockaddr_in*)inaddr.ai_addr;
+    if (inaddr.ai_socktype == SOCK_STREAM && addr->sin_family == AF_INET && addr->sin_port == 0) {
```

Comment:
Thanks for pointing the FTP server use case. Hope the new code matches what you were thinking.

Reply:
Thank you for addressing this (and other) problems. I am resolving this change request because the problem I flagged was addressed. I will open a new change request to discuss (and possibly adjust) _how_ that problem was addressed.